sound/soc/sof/ipc4-pcm.c

```diff
@@ -332,12 +332,12 @@ static int sof_ipc4_trigger_pipelines(struct snd_soc_component *component,
 	 * guaranteed for each fork independently.
```

Comment:
Not following. One one hand you say " This list is later traversed in the reverse order for triggering" implying the order of the widgets in the list does not matter. But then later, "we want both the DAI pipelines (0 and 1) to be triggered
after pipeline 2 has been started". I feel there is a missing sentence to explain why the latter requires the widget setup order to be modified? Is it just easier to implement this way, or is there a harder dependency?

What will happen with AEC-type topology and e.g. 6.5.0 kernel that does not have this patch?

Reply:
@kv2019i Ideally for us the firmware should be able to handle any order of pipeline trigger with the AEC-type topology but unfortunately that doesnt seem to be the case. If I keep the existing order of widget setup, the reference capture pipeline is the first one to be triggered and it results in an IPC timeout.